src/ServiceControl.Monitoring/Metrics/Raw/LongValueOccurenceSerializerDefinition.cs

```diff
@@ -24,33 +24,52 @@ public void Serialize(object message, Stream stream)
 
         public object[] Deserialize(Stream stream, IList<Type> messageTypes = null)
         {
-            var message = new LongValueOccurrences();
+            var reader = new BinaryReader(stream);
 
-            ToLongValueOccurrences(message, stream);
+            var version = reader.ReadInt64();
 
-            return new object[]
+            if (version == 1)
             {
-                message
-            };
-        }
+                var baseTicks = reader.ReadInt64();
+                var count = reader.ReadInt32();
 
-        static void ToLongValueOccurrences(LongValueOccurrences message, Stream stream)
-        {
-            var reader = new BinaryReader(stream);
+                if (count == 0)
+                {
+                    // no records, return empty
+                    return new object[0];
+                }
 
-            message.Version = reader.ReadInt64();
-            message.BaseTicks = reader.ReadInt64();
+                var messages = new List<object>(1); // usual case
+                var message = LeaseMessage(messages);
 
-            var count = reader.ReadInt32();
+                for (var i = 0; i < count; i++)
+                {
+                    var ticks = reader.ReadInt32();
+                    var value = reader.ReadInt64();
+                    var date = baseTicks + ticks;
 
-            message.Ticks = new int[count];
-            message.Values = new long[count];
+                    if (message.TryRecord(date, value) == false)
+                    {
+                        message = LeaseMessage(messages);
```

Comment:
I don't like that this is asymmetric. We lease new instances in the deserialization step and then release them in a handler. 

Would it be better to introduce a behavior to lease a new message, fill it, call the inner behavior, and then release it, all withing a single try-finally?